### PR TITLE
Resolved FB5005

### PR DESCRIPTION
**Pull request by seefo**
Should fix https://highfidelity.fogbugz.com/f/cases/5005/User-activity-events-can-be-out-of-order

Test Plan:
1. Run interface for a few minutes and ping @birarda once your session is complete

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/495/HighFidelity-Beta-PR10644-ada685a2b5d3b7f2bceef75f6c7d0e0a4e969de0.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/495/HighFidelity-Beta-PR10644-ada685a2b5d3b7f2bceef75f6c7d0e0a4e969de0.dmg

File: libraries/networking/src/UserActivityLogger.cpp [hunk 1]
```diff
@@ -20,6 +20,10 @@
 #include <DependencyManager.h>
 #include "AddressManager.h"
 
+UserActivityLogger::UserActivityLogger() : _timer() {
```
Comment (birarda) [inline libraries/networking/src/UserActivityLogger.cpp]: I think you can omit the `_timer()` here since that is the default constructor

**Comment by birarda** [comment]
This is waiting on backend changes before it can be QA'ed.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/500/HighFidelity-Beta-PR10644-fce3badd1d985eddc0e5fa1ee2bee19169eff8ee.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/500/HighFidelity-Beta-PR10644-fce3badd1d985eddc0e5fa1ee2bee19169eff8ee.dmg

**Comment by claudioshg** [comment]
testing

**Comment by claudioshg** [comment]
@birarda I opened the interface and have had it open for about 40 minutes. Should I close the interface?

**Comment by birarda** [comment]
@claudioshg I'll look at the data now, thanks!

**Comment by birarda** [comment]
QA Failed, the elapsed_ms is 0 for the duration of your session.

File: libraries/networking/src/UserActivityLogger.cpp [hunk 2]
```diff
@@ -42,6 +46,12 @@ void UserActivityLogger::logAction(QString action, QJsonObject details, JSONCall
     actionPart.setHeader(QNetworkRequest::ContentDispositionHeader, "form-data; name=\"action_name\"");
     actionPart.setBody(QByteArray().append(action));
     multipart->append(actionPart);
+
+    // Log the local-time that this event was logged
+    QHttpPart elapsedPart;
+    elapsedPart.setHeader(QNetworkRequest::ContentDispositionHeader, "form-data; name=\"elapsed_ms\"");
+    elapsedPart.setBody(QByteArray().append(_timer.elapsed()));
```
Comment (birarda) [inline libraries/networking/src/UserActivityLogger.cpp]: This is where it's getting messed up. Try `QString::number(_timer.elapsed).toLocal8Bit()`

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/545/HighFidelity-Beta-PR10644-06a5d0970c5916d7bcc78cc35a6e46cfed00f7c7.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/545/HighFidelity-Beta-PR10644-06a5d0970c5916d7bcc78cc35a6e46cfed00f7c7.dmg

**Comment by YuppyPlays** [comment]
Testing.

**Comment by birarda** [comment]
Sounds like @YuppyPlays crashed while testing this build - I'd like to double check on that before I called this a QA pass, but the data looks right.

**Comment by YuppyPlays** [comment]
Testing complete:

(Messaged Birarda as I started and as it ended to give the timeframe)

Not decided yet by dev if Pass or Fail. Please let me know if there is anything else I can do/provide to assist. :)

**Comment by birarda** [comment]
Your crash is in `interface!OffscreenQmlSurfaceCache::release` - unrelated to this PR.

**Comment by YuppyPlays** [comment]
@birarda ah excellent all is good then :D

**Pull request merged**